fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/api/client/render/fluid/v1/OpaqueFluid.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.render.fluid.v1;
+
+import net.minecraft.client.render.BackgroundRenderer;
+import net.minecraft.client.render.FogShape;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+
+/**
+ * Defines a non-transparent fluid with a fog.
```

Comment:
The javadoc should detail how to use the type (in this case, by implementing it)

Reply:
Now is clarified
```java
/**
 * <p>Defines a non-transparent fluid with a fog.</p>
 * <p>This interface must be implemented in each {@link FlowableFluid} class you want to specify fog parameters.</p>
 */
```